### DISCUSSION TITLE
标量过滤性能较差如何解决？

### DISCUSSION THREAD

**Discussion by hellwhj** [milvus-io/milvus#18152] [Q&A and General discussion]
**背景**
1.scheme的field共9个，包括id、vector（dim=10）和7个int32类型的标量：a、b0、b1、b2、c、d、s
2.数据量200w条，占用空间大约几百M，索引类型IVF_FLAT，nlist:1024，search时候nprobe:10，topK:500
3.我需要的标量过滤，expr举例：(a in [-395119942, -1109494664]) and ((b0 in [1490909981]) or (b1 in [1490909981]) or (b2 in [1490909981])) and (c <= 13.2) and (d >= 6.0) and (s == 2070491631)
**性能**
不用标量过滤：100ms
用上述标量过滤：250ms

**Comment by xiaofan-luan** [comment]
200w数据，应该不至于需要250ms，是2.0.2版本么，可以基于master或者稍后发布的2.1重新测试一下

**Reply by hellwhj** [reply]
[434-slow.log](https://github.com/milvus-io/milvus/files/9090896/434-slow.log)
[434-fast.log](https://github.com/milvus-io/milvus/files/9090898/434-fast.log)
这是一快一慢两个请求的log，两者的expr是一模一样的
